src/App.js

```diff
@@ -14,11 +14,102 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    visibleList: true,
+    visibleButton: false,
```

Comment:
```suggestion
```
Let's use conditional rendering instead of 'visibility' attribute. You can get rid of this 'visibleButton' variable and render this button only if the list is visible. Also, you should render your list only if 'visibleList' is true as well.

`{!visibleList && (<button>...</button>)}`
and
`{visibleList && (<ul>...</ul>)}`

By clicking on the button you just change 'visibleList' to 'true'

P.S: better to name it smth like 'listIsVisible'. It will be more meaningful

Reply:
Здається, знаю це, але на практиці важко використовувати, не одразу доходить, що можна так зробити
З назвами зміних в мене завжди проблеми, не завжди виходить влучно назвати